### DISCUSSION TITLE
Allow custom links to make POST requests( for API calls, custom scripts, etc)

### DISCUSSION THREAD

**Discussion by SaschaSchwarzK** [netbox-community/netbox#16200] [Ideas]
The custom link is a useful feature. For example you can add a button that opens the device in your monitoring system, that makes it very convenient for users.

But currently it is limited to GET requests. But if you want to use that to trigger any automation, even start a script on netbox itself you cannot do this as you need to make a POST request.

Another option would be to introduce a "Button" option to start a custom script directly from the model view page. But the option to select the request method seems much simpler to implement.

**Comment by Blusafur** [comment]
I would like all of these ideas as well. It would be nice if I could go to a device page in netbox, and have a link/button to execute a job via API in another system such as Ansible Tower/AWX.

Calling a script and passing a device variable such as name would be another great idea. That way I could call an API that has a token embedded in the call without a user being able to see it.

**Comment by reuben-thundergrid** [comment]
This would have been useful for us as we wanted to add explicit buttons in the UI for integrations with other systems as apposed to using event rules and webhooks.
